bin/oneoff/migrate_users_to_multi_auth.rb

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require_relative '../../dashboard/config/environment'
+
+SLICE_SIZE = 1_000
+
+def migrate_batches(limit=nil)
+  limit = limit.to_i if limit
+  unmigrated_users = User.where.not(provider: User::PROVIDER_MIGRATED)
```

Comment:
Do we need to `with_deleted` include soft-deleted users, or are we officially leaving those behind?

Reply:
My plan is to leave them behind, then wait 30 days after this script finishes to begin tearing out support for nonmigrated users to ensure that they are fully gone.